about/licenses/index.md

```diff
@@ -0,0 +1,88 @@
+---
+title: LLNL Software Licensing
+layout: default
+---
+
+## {{ page.title }}
+{: .page-header}
+
+*This page contains information meant for LLNL Employees with instructions for
+how to mark their source code projects. It is not intended to serve as legal
+advice*
+
+### Required Content
+
+All LLNL software must contain the following files and markings:
+
+#### 1. README file
+
+All projects shall have a `README.md` file at the root of the repository,
+including at least the following information:
+
+- "Getting Started" instructions
+- "Releasability" section, containing the `LLNL-CODE-XXXXXX` number
+
+#### 2. Disclaimer file
+
+At the root of every source code repository shall be the following text in a
+file named `DISCLAIMER`:
+
+```
+This work was produced under the auspices of the U.S. Department of Energy by
+Lawrence Livermore National Laboratory under Contract DE-AC52-07NA27344.
+
+This work was prepared as an account of work sponsored by an agency of the
+United States Government. Neither the United States Government nor Lawrence
+Livermore National Security, LLC, nor any of their employees makes any warranty,
+expressed or implied, or assumes any legal liability or responsibility for the
+accuracy, completeness, or usefulness of any information, apparatus, product, or
+process disclosed, or represents that its use would not infringe privately owned
+rights. Reference herein to any specific commercial product, process, or service
+by trade name, trademark, manufacturer, or otherwise does not necessarily
+constitute or imply its endorsement, recommendation, or favoring by the United
+States Government or Lawrence Livermore National Security, LLC. The views and
+opinions of authors expressed herein do not necessarily state or reflect those
+of the United States Government or Lawrence Livermore National Security, LLC,
+and shall not be used for advertising or product endorsement purposes.
+```
+
+#### 3. Source Code Headers
+
+All source code files in a repository shall have a comment at the top of each
+source code file with the following content:
+
+```
+Copyright (c) 2018, Lawrence Livermore National Security, LLC
```

Comment:
It should probably be made clear in our documentation *when* the lab expects contributors to hand over copyright, and what actions will cause the lab to assume they have done so.

When you contribute a file to a repository under a certain license, it's assumed that you grant people the right to *use* your file under that license, but not that you've granted copyright.  Generally you have to sign another document to hand over copyright (e.g., a suitably worded CLA).

In the Spack repo, most contributors happily label their work with *our* full copyright header, and a few others have contributed alternate headers that indicate that another organization retains copyright.  I am curious what lab legal would say about whether we actually now have the copyright on files where people used the LLNL copyright header.  If we do have copyright, it effectively means we can relicense the contribution.

Reply:
i've seen discussions previously as well on the applicability of these copyright headers, even though they are near ubiquitous in the industry.

Reply:
@jbjonesjr: can you send some pointers?

Reply:
no pointers, just various conversations. If i come across any, i'll send them your way

Reply:
What's your opinions on having a separate Contributor file?  Then the header/license refers to Contributors and references the separate file that can get updated without changing the legal license?  Some projects may want CLAs, others may be fine with a list in the repo...

Reply:
@Follett2: A contributor file is a lot of work for the authors of the code.  If you require a contributor file, it'll likely be wrong most of the time, and it won't tell you which author contributed what code.  Also, it seems like a waste when you can already get a list of contributors (and the lines they worked on) from the git history.  This is why we just link to the github [contributors](https://github.com/spack/spack/graphs/contributors) page for Spack.

I was really asking whether, when someone puts a copyright header on some code and contributes it to a project, the project can consider themselves to have copyright.

Reply:
I agree, I don't think a separate contributors file makes sense, due to the amount of work involved trying to maintain it. Instead, functionality such as GitHub's contributor pages are a much better way to go.